modules/subscription/main.tf

```diff
@@ -35,4 +35,33 @@ resource "azapi_resource" "subscription" {
     }
   })
   response_export_values = ["properties.subscriptionId"]
+  lifecycle {
+    ignore_changes = [
+      body
+    ]
+  }
+}
+
+resource "azapi_update_resource" "subscription_tags" {
+  count = var.subscription_alias_enabled && var.subscription_use_azapi ? 1 : 0
+
+  type        = "Microsoft.Resources/tags@2022-09-01"
+  resource_id = "/subscriptions/${jsondecode(azapi_resource.subscription[0].output).properties.subscriptionId}/providers/Microsoft.Resources/tags/default"
+  body = jsonencode({
+    properties = {
+      tags = var.subscription_tags
+    }
+  })
+}
+
+resource "azapi_resource_action" "subscription_rename" {
```

Comment:
On initial sub creation, is there an unnecessary POST request to rename the sub, even though the name hasn't changed?

Reply:
Yes there will be, similarly an unnecessary updating of the tags.